bin/action_runner.py

```diff
@@ -5,79 +5,96 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import argparse
 import contextlib
 import logging
 import os
 import subprocess
 import sys
 
-from tron import yaml
-
+import yaml
 
 log = logging.getLogger("tron.action_runner")
 
 
 STATUS_FILE = 'status'
 
 
-opener = open
-
-
 class StatusFile(object):
     """Manage a status file."""
 
     def __init__(self, filename):
         self.filename = filename
 
-    def write(self, command, proc):
-        with opener(self.filename, 'w') as fh:
-            yaml.dump(self.get_content(command, proc), fh)
-
-    def get_content(self, command, proc):
+    def get_content(self, run_id, command, proc):
         return {
+            'run_id':       run_id,
             'command':      command,
             'pid':          proc.pid,
             'return_code':  proc.returncode,
         }
 
     @contextlib.contextmanager
-    def wrap(self, command, proc):
-        self.write(command, proc)
-        try:
-            yield
-        finally:
-            self.write(command, proc)
-
-
-class NoFile(object):
```

Comment:
Did we not need this?

Reply:
this was the 'silent' failure mode that was used when the os.makedirs call failed. I opted to get rid of it and fail hard, rather than degraded.